src/Symfony/Component/Notifier/Bridge/Lox24/Webhook/Lox24RequestParser.php

```diff
@@ -33,28 +34,95 @@ protected function getRequestMatcher(): RequestMatcherInterface
     /**
      * @throws RejectWebhookException
      */
-    protected function doParse(Request $request, #[\SensitiveParameter] string $secret): ?SmsEvent
+    protected function doParse(Request $request, #[\SensitiveParameter] string $secret): SmsEvent|RemoteEvent|null
     {
         $payload = $request->request->all() ?? [];
         $name = $payload['name'] ?? null;
-        $data = $payload['data'] ?? [];
+        $data = $payload['data'] ?? null;
+        $id = $payload['id'] ?? null;
 
-        if ('sms.delivery' !== $name) {
-            throw new RejectWebhookException(400, 'Notification name is not \'sms.delivery\'');
+        $missingFields = [];
+        if (!$id) {
+            $missingFields[] = 'id';
+        }
+        if (!$name) {
+            $missingFields[] = 'name';
+        }
+        if (!$data) {
+            $missingFields[] = 'data';
+        }
+
+        if ($missingFields) {
+            throw new RejectWebhookException(406, \sprintf('The required fields "%s" are missing from the payload.', implode('", "', $missingFields)));
+        }
+
+        if (!\is_array($data)) {
+            throw new RejectWebhookException(406, 'The "data" field must be an array.');
+        }
+
+        return $this->createEventFromPayload($name, $data, $payload);
+    }
+
+    private function createEventFromPayload(string $eventName, array $data, array $payload): SmsEvent|RemoteEvent|null
+    {
+        return match ($eventName) {
+            'sms.delivery', 'sms.delivery.dryrun' => $this->createDeliveryEvent($data, $payload),
+            default => $this->createRemoteEvent($eventName, $payload),
+        };
+    }
+
+    private function createDeliveryEvent(array $data, array $payload): ?SmsEvent
+    {
+        if (!isset($data['id'])) {
```

Comment:
just for confirmation: we already checked that id is present in payload, and id now also has to be inside payload['data'], is that correct?

Reply:
I checked the doc, it's correct:
https://doc.lox24.eu/#tag/sms/operation/webhooks.sms-delivery-webhook